waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingServiceTest.java

```diff
@@ -445,6 +446,41 @@ public void panopticOperationsHandlerSetUgiDifferentTimeouts() throws Exception
     assertThat(result.contains("ugi"), is(true));
   }
 
+  @Test
+  public void panopticOperationsHandlerGetAllFunctions() throws Exception {
+    GetAllFunctionsResponse responsePrimary = new GetAllFunctionsResponse();
+    responsePrimary.addToFunctions(newFunction("db", "fn1"));
+    when(primaryDatabaseClient.get_all_functions()).thenReturn(responsePrimary);
+
+    when(metaStoreMappingFederated.getClient()).thenReturn(federatedDatabaseClient);
+    GetAllFunctionsResponse responseFederated = new GetAllFunctionsResponse();
+    responseFederated.addToFunctions(newFunction("db", "fn2"));
+    when(federatedDatabaseClient.get_all_functions()).thenReturn(responseFederated);
+
+    PanopticOperationHandler handler = service.getPanopticOperationHandler();
+    GetAllFunctionsResponse result = handler.getAllFunctions();
+    assertThat(result.getFunctionsSize(), is(2));
+    assertThat(result.getFunctions().get(0).getFunctionName(), is("fn1"));
+    assertThat(result.getFunctions().get(1).getFunctionName(), is("fn2"));
+  }
+
+  @Test
+  public void panopticOperationsHandlerGetAllFunctionsWithSlowConnection() throws Exception {
```

Comment:
There is a lot of duplication in these tests, I'll have a look and see if I can refactor it a bit.